src/app/modules/gui.cpp

```diff
@@ -634,6 +634,12 @@ bool CustomizedGuiManager::onProcessMessage(Message* msg)
             case KeyType::Command: {
               Command* command = key->command();
 
+              // If the current state has a wheel behavior, its needed to cancel it
+              // to avoid returning to the wheel behavior after dialog/command ends
+              if (current_editor &&
+                  current_editor->getState() &&
+                  current_editor->getState()->isWheelBehaviorState())
+                current_editor->getState()->onKeyUp(current_editor, static_cast<KeyMessage*>(msg));
```

Comment:
I think there are other states that handle onKeyUp(), but as you said it would be nice to have state-specific behavior in the same state (in this case DraggingValueState). Also calling onKeyUp() method directly is not desirable (it should only happen in the kKeyUpMessage handler of the Editor::onProcessMessage(), and in no other place).

Before modifying CustomizedGuiManager I would give a try to using a more generic solution like observing the BeforeCommandExecution signal of UIContext ([just like in DrawingState](https://github.com/aseprite/aseprite/blob/12e59b887f6d8e387a1d9d01662822a44efae87a/src/app/ui/editor/drawing_state.cpp#L75)).

Reply:
Thanks to everyone for the tips 🙌. I just pushed-forced the changes.